### PR TITLE
Preserve UnmanagedCallersOnlyAttribute on mono, the runtime needs it, and it can be removed when using --used-attrs-only.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml [hunk 1]
```diff
@@ -688,6 +688,10 @@
 			<field name="FirstChanceException"/>
 		</type>
 
+		<!-- mono_method_has_unmanaged_callers_only_attribute () -->
+		<!-- aot-compiler.c -->
```
Comment (akoeplinger) [inline src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml:692]: the method is in marshal.c

Comment (akoeplinger) [reply src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml:692]: I looked at the method, is the issue that `mono_class_try_get_unmanaged_callers_only_attribute_class` not gracefully handles the case where the attribute was linked out?

Comment (vargaz) [reply src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml:692]: The runtime/aot compiler needs this attribute, and the linker removes it from methods when using --used-attrs-only.

Comment (jeromelaban) [reply src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml:692]: The PInvokeTableGenerator also needs this: https://github.com/dotnet/runtime/blob/dbe6447aa29b14150b7c6dd43072cc75f0cdf013/tools-local/tasks/mobile.tasks/WasmAppBuilder/PInvokeTableGenerator.cs#L72

**Review by marek-safar — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Comment by MichalStrehovsky** [comment]
Is there a decoder ring for when these get added to CoreLib's XML and when do they get hardcoded into linker like https://github.com/mono/linker/pull/910?

(The `--used-attrs-only` option could be renamed to [`--break-the-app-subtly`](https://github.com/mono/linker/issues/952#issuecomment-601414891). When we finally put this option out of its misery and delete it, it would be preferable to have all of this special casing in the linker so that it disappears in one go.)

**Comment by vargaz** [comment]
I think hardcoding in the linker was the old approach, and adding it to xml is the new one.

**Pull request merged**